apps/api/src/pkg/keys/service.ts

```diff
@@ -591,7 +593,10 @@ export class KeyService {
 
     let remaining: number | undefined = undefined;
     if (data.key.remaining !== null) {
-      const limited = await this.usageLimiter.limit({ keyId: data.key.id });
+      const limited = await this.usageLimiter.limit({
+        keyId: data.key.id,
+        cost: req.remaining?.cost ?? 1,
```

Comment:
Nit: Can we move magic number to a constant then top of the file 😄

Reply:
@coderabbitai can you create a separate issue for this?

Reply:
the documentation said it can do this

Reply:
done